ballerina-to-openapi/src/main/java/io/ballerina/openapi/service/mapper/example/OpenAPIExampleMapperImpl.java

```diff
@@ -0,0 +1,295 @@
+                        headerParam.get(), semanticModel, diagnostics);
+                headerExampleMapper.setExample();
+                headerExampleMapper.setExamples();
+                break;
+            default:
+                break;
+        }
+    }
+
+    private void setExamplesForResourcePathParam(ResourcePathParameterNode parameterNode, Operation operation) {
+        Optional<Symbol> parameterSymbolOpt = semanticModel.symbol(parameterNode);
+        if (parameterSymbolOpt.isEmpty() ||
+                !(parameterSymbolOpt.get() instanceof PathParameterSymbol parameterSymbol)) {
+            return;
+        }
+
+        Optional<Parameter> pathParam = operation.getParameters().stream()
+                .filter(parameter -> parameter.getIn().equals(PATH) &&
+                        parameter.getName().equals(unescapeIdentifier(parameterSymbol.getName().get())))
+                .findFirst();
+        if (pathParam.isEmpty()) {
+            return;
+        }
+
+        ExamplesMapper pathExampleMapper = new PathExampleMapper(parameterSymbol, pathParam.get(),
+                semanticModel, diagnostics);
+        pathExampleMapper.setExample();
+        pathExampleMapper.setExamples();
+    }
+
+    private static ParameterType getParameterType(ParameterSymbol parameterSymbol, SemanticModel semanticModel) {
+        TypeSymbol parameterTypeSymbol = parameterSymbol.typeDescriptor();
+        if (!parameterTypeSymbol.subtypeOf(semanticModel.types().ANYDATA)) {
+            return ParameterType.OTHER;
+        }
+
+        List<AnnotationSymbol> httpAnnotations = parameterSymbol.annotations().stream()
+                .filter(annotation -> annotation.typeDescriptor().isPresent() &&
+                        isHttpPackageAnnotationTypeDesc(annotation.typeDescriptor().get()))
+                .toList();
+        if (httpAnnotations.isEmpty()) {
+            return ParameterType.QUERY;
+        }
+
+        for (AnnotationSymbol annotation : httpAnnotations) {
+            TypeSymbol annotationType = annotation.typeDescriptor().get();
+            if (isSubTypeOfHttpType(annotationType, HTTP_PAYLOAD_TYPE, semanticModel)) {
+                return ParameterType.PAYLOAD;
+            }
+            if (isSubTypeOfHttpType(annotationType, HTTP_HEADER_TYPE, semanticModel)) {
+                return ParameterType.HEADER;
+            }
+            if (isSubTypeOfHttpType(annotationType, HTTP_QUERY_TYPE, semanticModel)) {
+                return ParameterType.QUERY;
+            }
+        }
+        return ParameterType.QUERY;
+    }
+
+    private static boolean isHttpPackageAnnotationTypeDesc(TypeSymbol typeSymbol) {
+        Optional<ModuleSymbol> module = typeSymbol.getModule();
+        if (module.isEmpty()) {
+            return false;
+        }
+        ModuleID id = module.get().id();
+        return id.orgName().equals(BALLERINA) && id.moduleName().startsWith(HTTP);
```

Comment:
can there be cases where this gets null?

Reply:
I don't think so. All the Ballerina packages should have a `orgName` and `packageName`. The `moduleName` points to the package module name, and if the module is the default(root) module then `packageName` and `moduleName` are same